products/sle/main.pm

```diff
@@ -57,19 +57,19 @@ sub is_update_test_repo_test {
 
 sub cleanup_needles {
     remove_common_needles;
-    if (get_var('VERSION', '') ne '12') {
+    if ((get_var('VERSION', '') ne '12') && (!get_var('BASE_VERSION', '') eq '12')) {
```

Comment:
can you use the same syntax for both comparisons, i.e.

```perl
(get_var('VERSION', '') ne '12') && (get_var('BASE_VERSION', '') ne '12')
```
or is my logic wrong here?